test/octree/test_octree.cpp

```diff
@@ -962,6 +962,414 @@ TEST (PCL, Octree_Pointcloud_Iterator_Test)
   // check if every branch node and every leaf node has been visited
   ASSERT_EQ (octreeA.getLeafCount (), leafNodeCount);
   ASSERT_EQ (octreeA.getBranchCount (), branchNodeCount);
+
+
+  // breadth-first at a given depth iterator test
+
+  lastDepth = 0;
+  branchNodeCount = 0;
+  leafNodeCount = 0;
+
+  leafNodeVisited = false;
+
+  OctreePointCloud<PointXYZ>::FixedDepthIterator bfdIt;
+  const OctreePointCloud<PointXYZ>::FixedDepthIterator bfdIt_end = octreeA.fixed_depth_end ();
+
+  for (unsigned int depth = 0; depth <= octreeA.getTreeDepth (); ++depth)
+  {
+    for (bfdIt = octreeA.fixed_depth_begin (depth); bfdIt != bfIt_end; ++bfdIt)
+    {
+      unsigned int current_depth = bfdIt.getCurrentOctreeDepth ();
+      ASSERT_TRUE (lastDepth <= current_depth);
+      ASSERT_TRUE (depth == current_depth);
+      lastDepth = current_depth;
+
+      if (bfdIt.isBranchNode ())
+      {
+        branchNodeCount++;
+        ASSERT_FALSE (leafNodeVisited);
+      }
+
+      if (bfdIt.isLeafNode ())
+      {
+        leafNodeCount++;
+        leafNodeVisited = true;
+      }
+    }
+  }
+
+  // check if every branch node and every leaf node has been visited
+  ASSERT_EQ (octreeA.getLeafCount (), leafNodeCount);
+  ASSERT_EQ (octreeA.getBranchCount (), branchNodeCount);
+
+  // Test the operator==() and operator!=()
```

Comment:
These tests to your iterator equality/inequality operators should be moved to their own test cases.

Reply:
Done